opsmngr/agent_api_keys.go

```diff
@@ -0,0 +1,91 @@
+package opsmngr
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	atlas "github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+)
+
+const (
+	agentAPIKeysBasePath = "groups/%s/agentapikeys"
+)
+
+type AgentAPIKeysService interface {
+	Create(context.Context, string, AgentAPIKeysRequest) (*AgentAPIKey, *atlas.Response, error)
+	List(context.Context, string) (*[]AgentAPIKey, *atlas.Response, error)
+	Delete(context.Context, string, string) (*atlas.Response, error)
+}
+
+type AgentAPIKeysServiceOp struct {
+	Client atlas.RequestDoer
+}
+
+type AgentAPIKey struct {
+	ID            string  `json:"_id"`
+	Key           string  `json:"key"`
+	Desc          string  `json:"desc"`
+	CreatedTime   int64   `json:"createdTime"`
+	CreatedUserID *string `json:"createdUserId,omitempty"`
+	CreatedIPAddr *string `json:"createdIpAddr,omitempty"`
+	CreatedBy     string  `json:"createdBy"`
+}
+
+type AgentAPIKeysRequest struct {
+	Desc string `json:"desc"`
+}
+
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/agentapikeys/create-one-agent-api-key/
+func (s *AgentAPIKeysServiceOp) Create(ctx context.Context, projectID string, agent AgentAPIKeysRequest) (*AgentAPIKey, *atlas.Response, error) {
+	path := fmt.Sprintf(agentAPIKeysBasePath, projectID)
+
+	req, err := s.Client.NewRequest(ctx, http.MethodPost, path, agent)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(AgentAPIKey)
+	resp, err := s.Client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return root, resp, err
+}
+
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/agentapikeys/get-all-agent-api-keys-for-project/
+func (s *AgentAPIKeysServiceOp) List(ctx context.Context, projectID string) (*[]AgentAPIKey, *atlas.Response, error) {
```

Comment:
List return an array of Agent of this format:
```json
[ {
  "_id" : "5e6f70ff917b225d8c2198ae",
  "createdBy" : "PUBLIC_API",
  "createdIpAddr" : "77.107.233.162",
  "createdTime" : 1584361727223,
  "createdUserId" : "5e6b84fd917b225d8c144143",
  "desc" : "Agent API Key for this project",
  "key" : "****************************8065"
}, {
  "_id" : "5e661ab0917b220fbd8bb713",
  "createdBy" : "USER",
  "createdIpAddr" : "77.107.233.162",
  "createdTime" : 1583749808084,
  "createdUserId" : "5e66185d917b220fbd8bb4c9",
  "desc" : "Initial API Key",
  "key" : "****************************a407"
} ]%
```

I tried  to use  AgentAPIKeys (I did the following attempts:  

1. without Links and Tot
2. without Links, Tot and json
3. with Links, Tot and without json)

But I always got the following error: **json: cannot unmarshal array into Go value of type opsmngr.AgentAPIKeys**

Putting []Agent fixed the issue.

Reply:
it looks like the API is not paginated so this is (kind of) expected